sdk/storage/Azure.Storage.Common/tests/StructuredMessageDecodingStreamTests.cs

```diff
@@ -102,6 +127,136 @@ public StructuredMessageDecodingStreamTests(ReadMethod method)
             Assert.That(new Span<byte>(decodedData).SequenceEqual(originalData));
         }
 
+        [Test]
+        public void BadStreamBadVersion()
+        {
+            byte[] originalData = new byte[1024];
+            new Random().NextBytes(originalData);
+            byte[] encodedData = StructuredMessageHelper.MakeEncodedData(originalData, 256, Flags.StorageCrc64);
+
+            encodedData[0] = byte.MaxValue;
+
+            Stream decodingStream = new StructuredMessageDecodingStream(new MemoryStream(encodedData));
+            Assert.That(async () => await CopyStream(decodingStream, Stream.Null), Throws.InnerException.TypeOf<InvalidDataException>());
+        }
+
+        [Test]
+        public async Task BadSegmentCrcThrows()
+        {
+            const int segmentLength = 256;
+            Random r = new();
+
+            byte[] originalData = new byte[2048];
+            r.NextBytes(originalData);
+            byte[] encodedData = StructuredMessageHelper.MakeEncodedData(originalData, segmentLength, Flags.StorageCrc64);
+
+            const int badBytePos = 1024;
+            encodedData[badBytePos] = (byte)~encodedData[badBytePos];
+
+            MemoryStream encodedDataStream = new(encodedData);
+            Stream decodingStream = new StructuredMessageDecodingStream(encodedDataStream);
+
+            //
+            const int copyBufferSize = 4;
+            bool caught = false;
+            try
+            {
+                await CopyStream(decodingStream, Stream.Null, copyBufferSize);
+            }
+            catch (CopyStreamException ex)
+            {
+                caught = true;
+                Assert.That(ex.TotalCopied, Is.LessThanOrEqualTo(badBytePos));
+            }
+            Assert.That(caught);
+        }
+
+        [Test]
+        public void BadStreamCrcThrows()
+        {
+            const int segmentLength = 256;
+            Random r = new();
+
+            byte[] originalData = new byte[2048];
+            r.NextBytes(originalData);
+            byte[] encodedData = StructuredMessageHelper.MakeEncodedData(originalData, segmentLength, Flags.StorageCrc64);
+
+            encodedData[originalData.Length - 1] = (byte)~encodedData[originalData.Length - 1];
+
+            Stream decodingStream = new StructuredMessageDecodingStream(new MemoryStream(encodedData));
+            Assert.That(async () => await CopyStream(decodingStream, Stream.Null), Throws.InnerException.TypeOf<InvalidDataException>());
+        }
+
+        [Test]
+        public void BadStreamWrongStreamFooter()
```

Comment:
Am I crazy or is this test the same as the one above?

Reply:
i may be the crazy one 😅